models/tasks.js

```diff
@@ -631,6 +632,40 @@ const updateTaskStatus = async () => {
   }
 };
 
+const updateOrphanTasksStatus = async (lastOrphanTasksFilteration) => {
+  const lastTimestamp = Number(lastOrphanTasksFilteration);
+  try {
+    const users = [];
+    const currentTimeStamp = Date.now();
+
+    const usersQuerySnapshot = await userModel
+      .where("roles.in_discord", "==", false)
+      .where("updated_at", ">=", lastTimestamp)
+      .where("updated_at", "<=", currentTimeStamp)
+      .get();
+
+    usersQuerySnapshot.forEach((user) => users.push({ ...user.data(), id: user.id }));
+
+    let orphanTasksUpdatedCount = 0;
+
+    for (const user of users) {
+      const tasksQuerySnapshot = await tasksModel
+        .where("assigneeId", "==", user.id)
+        .where("status", "not-in", ["COMPLETED", "BACKLOG"])
+        .get();
+      tasksQuerySnapshot.forEach(async (taskDoc) => {
+        orphanTasksUpdatedCount++;
+        await tasksModel.doc(taskDoc.id).update({ status: "BACKLOG" });
+      });
+    }
+
+    return { orphanTasksUpdatedCount };
+  } catch (error) {
+    logger.error("Error marking tasks as backlog:", error);
+    throw error;
+  }
```

Comment:
fetching all users and then iterating over them to find orphan tasks might not be efficient, especially with a large dataset. You can optimise the database query to directly fetch orphan tasks based on a predefined condition.

Reply:
We are not fetching all the users we are fetching only users who's user documents are updated after last run of the orphan Tasks filtration and also user's who are not it discord `in_discord: false`. 

only running this for first time will have a lot of users. Later it will be efficient.

Reply:
>  You can optimise the database query to directly fetch orphan tasks based on a predefined condition.

What will be other predefined condition? Orphan tasks are tasks which are not completed and the assignee is not anymore in our discord server.

Reply:
I see that the predefined condition already encompasses the necessary criteria for identifying orphan tasks. Therefore, there might not be another predefined condition needed.
thank you for clarifying